example/particle_leapfrog/particle_leapfrog.cpp

```diff
@@ -0,0 +1,455 @@
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#include "particle_leapfrog.hpp"
+#include "Kokkos_CopyViews.hpp"
+#include "Kokkos_HostSpace.hpp"
+#include "basic_types.hpp"
+#include "config.hpp"
+#include "globals.hpp"
+#include "interface/update.hpp"
+#include "kokkos_abstraction.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <limits>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+// *************************************************//
+// redefine some internal parthenon functions      *//
+// *************************************************//
+namespace particles_example {
+
+Packages_t ProcessPackages(std::unique_ptr<ParameterInput> &pin) {
+  Packages_t packages;
+  packages.Add(particles_example::Particles::Initialize(pin.get()));
+  return packages;
+}
+
+// *************************************************//
+// define the "physics" package particles_package, *//
+// which includes defining various functions that  *//
+// control how parthenon functions and any tasks   *//
+// needed to implement the "physics"               *//
+// *************************************************//
+
+namespace Particles {
+
+std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
+  auto pkg = std::make_shared<StateDescriptor>("particles_package");
+
+  Real cfl = pin->GetOrAddReal("Particles", "cfl", 0.3);
+  pkg->AddParam<>("cfl", cfl);
+
+  auto write_particle_log =
+      pin->GetOrAddBoolean("Particles", "write_particle_log", false);
+  pkg->AddParam<>("write_particle_log", write_particle_log);
+
+  std::string swarm_name = "my particles";
+  Metadata swarm_metadata;
+  pkg->AddSwarm(swarm_name, swarm_metadata);
+  Metadata real_swarmvalue_metadata({Metadata::Real});
+  pkg->AddSwarmValue("id", swarm_name, Metadata({Metadata::Integer}));
+  pkg->AddSwarmValue("vx", swarm_name, real_swarmvalue_metadata);
+  pkg->AddSwarmValue("vy", swarm_name, real_swarmvalue_metadata);
+  pkg->AddSwarmValue("vz", swarm_name, real_swarmvalue_metadata);
+
+  pkg->EstimateTimestepBlock = EstimateTimestepBlock;
+
+  return pkg;
+}
+
+AmrTag CheckRefinement(MeshBlockData<Real> *rc) { return AmrTag::same; }
```

Comment:
Is this actually needed? I think this should be the default, no?

Reply:
Correct. Not needed and removed.